core/spec/models/spree/order_spec.rb

```diff
@@ -1600,6 +1600,40 @@ def generate
     it 'creates at least one new shipment for the order' do
       expect { subject }.to change { order.shipments.count }.by 1
     end
+
+    context "with a custom inventory unit builder" do
+      before do
+        class TestInventoryUnitBuilder
+          def initialize(order)
+          end
+
+          def missing_units_for_line_item(line_item)
+            []
+          end
+        end
+
+        # This return value is stubbed this way, so that it can access the
+        # arbitrary_inventory_unit let variable.
+        allow_any_instance_of(TestInventoryUnitBuilder)
+          .to receive(:missing_units_for_line_item)
+          .and_return([arbitrary_inventory_unit])
+
+        test_stock_config = Spree::Core::StockConfiguration.new
+        test_stock_config.inventory_unit_builder_class = TestInventoryUnitBuilder.to_s
+        stub_spree_preferences(stock: test_stock_config)
+
+        order.shipments.first.update!(created_at: 2.years.ago)
```

Comment:
What about creating a targeted order (instead of the one defined "globally")? It'd make intentions more apparent, and we wouldn't need this workaround to be sure we select the correct shipment.

Reply:
I'm not sure what you're suggesting here. Could you clarify?

Reply:
Hey @jarednorman, I'm still confused about having to set `2.years.ago` here :thinking: 

Wouldn't it be possible something like that?

```ruby
# A single shipment with no inventory unit
let(:order) { create(:order, shipments: create(:shipment, inventory_units: [])) }
before do
 # Everything except the "order.shipments.first.update!(created_at: 2.years.ago)" line
end
# ...
it "relies on the custom builder" do
# replacing the second except with:
expect(order.shipments.first.inventory_units).to contain_exactly arbitrary_inventory_unit
end
```